battlesim.py

```diff
@@ -40,10 +40,37 @@ def isDps(self):
             return False
 
 
-class Fleet:
-    def __init__(self, ships):
-        self.ships = ships
+# Ship(dps, high_slot_hps, mid_slot_hps, ehp, role, timer)
+# the number is divided by resist
 
+
+def dps():
+    return Ship(1500, 0, 387, 250000, 'Dps', 0)
+
+
+def healer():
+    return Ship(0, 3377, 309, 100000, 'Healer', 0)
+
+
+def tanker():
+    return Ship(300, 0, 0, 250000, 'Tanker', 0)
+
+class Fleet:
+    # type is the number of ship in order of healer, tank, dps, pm = primary mode is the order that ship gets attackted
+    def __init__(self, pm, t, h, d):
+        self.ships = []
+        for o in pm:
+            if (o == 2):
+                for i in range(h):
+                    self.ships.append(healer())
+            if (o == 1):
+                for i in range(t):
+                    self.ships.append(tanker())
+            if (o == 3):
+                for i in range(d):
+                    self.ships.append(dps())
+
```

Comment:
Make Fleet class more OO, you can define this kind of utility functions outside of Fleet class.